morpheus/stages/general/nonlinear_modules_stage.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import typing
+
+import mrc
+
+from morpheus.config import Config
+from morpheus.pipeline.stage import Stage
+from morpheus.pipeline.stream_pair import StreamPair
+from morpheus.utils.module_utils import load_module
+
+logger = logging.getLogger(__name__)
+
+
+class NonLinearModulesStage(Stage):
+    """
+    Loads an existing, registered, MRC SegmentModule and wraps it as a Morpheus Stage.
+
+    Parameters
+    ----------
+    c : `morpheus.config.Config`
+        Pipeline configuration instance.
+    module_config : typing.Dict
+        Module configuration.
+    input_port_name : str
+        Name of the input port for the registered module.
+    output_port_name_prefix : str
+        Prefix name of the output ports for the registered module.
+    output_port_count : str
+        Number of output ports for the registered module.
+    input_type : default `typing.Any`
+        The stage acceptable input type.
+    output_type : default `typing.Any`
+        The output type that the stage produces.
```

Comment:
All of these parameters should be reduced down to `input_port_pairs` and `output_port_pairs` which are `typing.List[typing.Tuple[str, typing.Type]]`. 

The current design limits the number of input ports to 1 and limits the output types to all be the same when this really isnt a needed restriction.